packages/devtools_app/lib/src/shared/utils/utils.dart

```diff
@@ -185,17 +185,39 @@ extension IsKeyType on KeyEvent {
   bool get isKeyDownOrRepeat => this is KeyDownEvent || this is KeyRepeatEvent;
 }
 
+/// A helper class for [Timer] functionality, where the callbacks are debounced.
+class Debouncer {
```

Comment:
why do we want to use this new class in favor of PeriodicDebouncer? I would think that handling all callbacks no matter what could lead us to doing unnecessary work, which is what the PeriodicDebouncer avoids

Reply:
Chatted offline - PeriodicTimer is useful for polling where callback will be the same each time. 

The implementation of Debouncer only calls the last callback given after the debounce duration, so it won't do unnecessary work. Added a comment to clarify.